mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/dao/DefaultDaoReturnTypeParser.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.mapper.processor.dao;
+
+import com.datastax.oss.driver.api.core.MappedAsyncPagingIterable;
+import com.datastax.oss.driver.api.core.PagingIterable;
+import com.datastax.oss.driver.api.core.cql.AsyncResultSet;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.datastax.oss.driver.api.core.cql.Row;
+import com.datastax.oss.driver.internal.mapper.processor.ProcessorContext;
+import java.util.Optional;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+
+public class DefaultDaoReturnTypeParser implements DaoReturnTypeParser {
+
+  private final ProcessorContext context;
+
+  public DefaultDaoReturnTypeParser(ProcessorContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public DaoReturnType parse(TypeMirror returnTypeMirror) {
```

Comment:
I didn't try to make any part of this method reusable, but it might be possible without being too messy. I'll give it a try.

Reply:
Yes, I think ti will be useful to make it easily extensible for clients.
If they will decide to plug custom logic overriding this:
```
@Override
  public DaoReturnTypeParser getDaoReturnTypeParser() {
    return daoReturnTypeParser;
  }
```
This is possible that the client will want to override handling only one specific `getKind()` so I think it will be useful to handle rest of the kinds by simply calling `super` instead of copy-pasting this method